src/components/Radio/Radio.tsx

```diff
@@ -1,66 +1,157 @@
 import * as React from 'react'
 import * as PropTypes from 'prop-types'
 
-import { createHTMLInput, customPropTypes, UIComponent } from '../../lib'
+import {
+  createHTMLInput,
+  customPropTypes,
+  AutoControlledComponent,
+  createShorthandFactory,
+} from '../../lib'
 import Label from '../Label'
-import { Extendable, ReactChildren } from '../../../types/utils'
+import { ComponentEventHandler, Extendable, ReactChildren } from '../../../types/utils'
 import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import Icon from '../Icon/Icon'
 
 export interface IRadioProps {
   as?: any
+  checked?: boolean
   children?: ReactChildren
   className?: string
+  disabled?: boolean
   label?: string
+  name?: string
+  onChange?: ComponentEventHandler<IRadioProps>
   type?: string
   styles?: IComponentPartStylesInput
+  value?: string | number
   variables?: ComponentVariablesInput
 }
 
 /**
  * @accessibility
  * This is shown at the top.
  */
-class Radio extends UIComponent<Extendable<IRadioProps>, any> {
+class Radio extends AutoControlledComponent<Extendable<IRadioProps>, any> {
+  static create: Function
+
   static displayName = 'Radio'
 
   static className = 'ui-radio'
 
   static propTypes = {
     as: customPropTypes.as,
 
+    /** Whether or not radio is checked. */
+    checked: PropTypes.bool,
+
     /** Child content * */
     children: PropTypes.node,
 
     /** Additional classes. */
     className: PropTypes.string,
 
+    /** Initial checked value. */
+    defaultChecked: PropTypes.bool,
```

Comment:
this one is not used anywhere - should we delete it?

Reply:
also, it seems that this one is rather pertains to `RadioGroup` concept, rather than to individual `Radio`

Reply:
Every auto-controlled prop must have a default prop associated to it, that is why we have it. I may need to provide example usage of it.

Reply:
@kuzhelov , I added the usage of the default checked value. This value is intended to be used as a initial check value of the radio button. As the component is auto-controlled, the user should be able to initially set the value of the checked property. Please comment if you have some more doubts on this.

Reply:
ok - but in that case let me share what my concerns are about. Now we have this `defaultChecked` to be defined in prop types, but there is no such a property in the component's TS interface - with that it, essentially, means that me as a client is not able to specify any value for it in the TS code. Is it intentional?

Reply:
Sorry, that is my fault, the defaultChecked prop **SHOULD** be provide in the TS interface. Thanks for the catch!